src/main/java/com/hollingsworth/arsnouveau/common/crafting/recipes/ArmorUpgradeRecipe.java

```diff
@@ -53,14 +54,6 @@ public boolean doesReagentMatch(ApparatusRecipeInput input, Level level, @org.je
         return armorPerkHolder.getTier() == (tier - 1);
     }
 
-    @Override
-    public boolean matches(ApparatusRecipeInput input, Level level, @org.jetbrains.annotations.Nullable Player player) {
-        if(this.pedestalItems().size() != input.pedestals().size()){
-            return false;
-        }
-        return this.doesReagentMatch(input, level, player);
```

Comment:
Adding the match check here should be the only needed change, modifying the other files reintroduces other issues, and pedestal checking is not faster for anything other than enchantment specific recipes